vllm/model_executor/models/jina_embeddings_v4.py

```diff
@@ -0,0 +1,363 @@
+            
+            # Validate lengths match
+            if len(token_ids_list) != len(prompt_lens):
+                logger.error(f"Mismatch: {len(token_ids_list)} sequences vs {len(prompt_lens)} lengths")
+                return self._base_pooler(hidden_states, pooling_metadata)
+            
+            # Apply optimized pooling
+            try:
+                pooled_data = self._apply_vision_pooling_optimized(
+                    hidden_states, token_ids_list, prompt_lens
+                )
+            except RuntimeError as e:
+                if "out of memory" in str(e).lower():
+                    logger.warning("OOM during pooling, falling back to sequential processing")
+                    # Process sequences one by one to reduce memory
+                    pooled_data = []
+                    for i in range(len(token_ids_list)):
+                        single_pooled = self._apply_vision_pooling_pytorch(
+                            hidden_states,
+                            [token_ids_list[i]],
+                            prompt_lens[i:i+1]
+                        )
+                        pooled_data.extend(single_pooled)
+                else:
+                    raise
+            
+            # Build output
+            pooled_outputs = [
+                PoolingSequenceGroupOutput(data) for data in pooled_data
+            ]
+            
+            # Record metrics
+            if self.observability_config:
+                elapsed_ms = (time.time() - start_time) * 1000
+                self._pooling_time_ms += elapsed_ms
+                self._pooling_count += 1
+                
+                if self._pooling_count % 100 == 0:
+                    avg_time = self._pooling_time_ms / self._pooling_count
+                    logger.debug(f"Average pooling time: {avg_time:.2f}ms")
+            
+            return PoolerOutput(outputs=pooled_outputs)
+            
+        except Exception as e:
+            logger.error(f"Error in pooler: {type(e).__name__}: {e}")
+            # Graceful degradation to base pooler
+            logger.info("Falling back to base pooler due to error")
+            return self._base_pooler(hidden_states, pooling_metadata)
+        
+        finally:
+            # Rely on Python's garbage collector for releasing tensors.
+            # torch.cuda.empty_cache() is a blocking and expensive operation
+            # that should be used sparingly.
+            pass
+
+    def load_weights(self, weights: Iterable[tuple[str, torch.Tensor]]):
+        """Load weights with validation and error handling."""
+        try:
+            loader = AutoWeightsLoader(self)
+            loaded_weights = loader.load_weights(weights, mapper=self.weight_mapper)
+            logger.info(f"Successfully loaded {len(loaded_weights)} weight tensors")
+            return loaded_weights
+        except Exception as e:
+            logger.error(f"Error loading weights: {e}")
+            raise
```

Comment:
```suggestion
        loader = AutoWeightsLoader(self)
        loaded_weights = loader.load_weights(weights, mapper=self.weight_mapper)
        logger.info(f"Successfully loaded {len(loaded_weights)} weight tensors")
        return loaded_weights
```
Please clean up the `try: ... except: ...` statement used in debug.